### ISSUE TITLE
repull sample collection date from GOLD 

### ISSUE THREAD

**Issue by aclum** [microbiomedata/issues#200]
**Is your feature request related to a problem? Please describe.**
There was bug in the GOLD API and we are missing sample collection date for at least GROW, Bioscales, Lake Mendota, NEON, and AGFMA.
**Describe the solution you'd like**
Identify the full collection of studies which have GOLD biosample identifier which are missing slot collection_date for class Biosample and update the records in mongo prod.

**Describe alternatives you've considered**

leave records incomplete

**Acceptance Criteria**
Collection date is populated on biosample pages on the data portal for biosamples from the above listed studies (ex https://data-dev.microbiomedata.org/details/sample/nmdc:bsm-11-pnb58t51)

**Comment by ssarrafan** [comment]
@aclum should this be moved to the current sprint?  And who should it be assigned to?

**Comment by aclum** [comment]
@sujaypatil96 is working on this sprint.

**Comment by sujaypatil96** [comment]
Submitted changesheets to correct biosample metadata records for GROW, Lake Mendota and NEON studies.

@aclum started a conversation with Stan to confirm if we should be pulling in the collection date from the spreadsheet he provided or from GOLD, because there's a mismatch between the two.

**Comment by sujaypatil96** [comment]
In a separate problem the AGFMA study on GOLD does not have the MIxS ENVO triad field values, which are _required_ fields in the schema. populated anymore because of which the pipeline isn't able to create new JSON artifacts for that study.

**Comment by sujaypatil96** [comment]
Hm, I rechecked the AGFMA biosample records on GOLD and now they seem to have the MIxS ENVO triad values populated. It didn't have the triad when I checked yesterday. Rerunning the pipeline now.

**Comment by sujaypatil96** [comment]
Maybe some of the AGFMA records don't have the MIxS ENVO triad values populated? I say this because the pipeline is still erroring out.

**Comment by aclum** [comment]
This is true for some of the study records but should not be true for any of the product types NMDC cares about. This should be handled by the product filtering.

**Comment by sujaypatil96** [comment]
Sounds good. The filtering feature, i.e., filter by biosample ids and sequencing project ids has not been integrated into the pipeline after we moved the pipeline codebase to the `nmdc-runtime` infrastructure. I can work on integrating that feature into the codebase this sprint, and then create the JSON artifact for the AGFMA study once that feature has been merged into `nmdc-runtime` if that works?

**Comment by aclum** [comment]
Yes, that sounds reasonable. Please do that after https://github.com/microbiomedata/sample-annotator/issues/115 since that is higher priority.

**Comment by aclum** [comment]
Moving to the next sprint since this is still in progress

**Comment by sujaypatil96** [comment]
Made a changesheet to update the `collection_date` values for all bioscales records: https://docs.google.com/spreadsheets/d/14TIg0bGfCzMAfZ1MEiK4REGZyNbuyUeksz2bMReBGTk/edit?usp=sharing

**Comment by sujaypatil96** [comment]
I will submit the above changesheet once @aclum confirms the correctness of the updates seeking to be made.

**Comment by sujaypatil96** [comment]
Submitted the changesheet from the above message using the API. I randomly checked a couple of biosample records from the bioscales study, and they all seem to have updated collection date values.

**Comment by aclum** [comment]
@sujaypatil96 is AGFMA still outstanding pending the updates to nmdc-runtime to incorporate a list of GOLD sample IDs?

**Comment by sujaypatil96** [comment]
@aclum yes, the collection dates for AGFMA haven't been applied yet. I will need to make another changesheet for that. If this is urgent, I can make a quick and dirty fix on my local nmdc-runtime and exclude the biosamples from the linked spreadsheet to regenerate the JSON for AGFMA.

If not, I can make the addition of inclusion and exclusion feature a priority for the coming sprint and push it up to nmdc-runtime upstream.

**Comment by aclum** [comment]
This can wait until the next sprint or the one after.

**Comment by ssarrafan** [comment]
@aclum @sujaypatil96 is this still a priority? No updates during this sprint.  Should it be moved to the next sprint?

**Comment by sujaypatil96** [comment]
Yes, it should be moved to the next sprint. The process needs to be repeated on AGFMA metadata still.

**Comment by aclum** [comment]
With work for NEON it is possible this should go to the following sprint (July 3 -14)

**Comment by ssarrafan** [comment]
This issue hasn't been updated in over a month. I'm removing from sprint and adding the backlog label.  @aclum @sujaypatil96

**Comment by ssarrafan** [comment]
@aclum @sujaypatil96 can this issue be closed?  It's been in many sprints.

**Comment by aclum** [comment]
This is still needed but is low priority.

**Comment by ssarrafan** [comment]
@aclum @sujaypatil96 can we close this one?  Or slot it into a sprint if really needed?

**Comment by sujaypatil96** [comment]
I just looked in the database for the biosample in the acceptance criteria and it has a `collection_date`. I assume we populated it via a changesheet as some point, but we'll need to spot check some biosample records across the studies in question to see if they have `collection_date` populated.

**Comment by ssarrafan** [comment]
Moving this to the runtime repo and board. Backlog grooming Feb 2025.
